test_scripts/Polices/External_UCS/021_ATF_N_TC_PTU_with_disallowed_by_external_consent_entities_off_struct_with_invalid_type_of_params.lua

```diff
@@ -0,0 +1,213 @@
+-- Requirement summary:
+-- [Policies] External UCS: PTU with "disallowed_by_external_consent_entities_off" struct with invalid type of params
+--
+-- Description:
+-- In case:
+-- SDL receives PolicyTableUpdate with “disallowed_by_external_consent_entities_off:
+-- [entityType: <any_type_except_Integer>, entityId: <any_type_except_Integer>]”
+-- -> of "<functional grouping>" -> from "functional_groupings" section
+-- SDL must:
+-- a. consider this PTU as invalid
+-- b. do not merge this invalid PTU to LocalPT.
+--
+-- Steps:
+-- 1. Register app
+-- 2. Activate app
+-- 3. Perform PTU (make sure 'disallowed_by_external_consent_entities_off' section is defined
+-- with invalid parameters)
+-- 4. Verify status of update
+--
+-- Expected result:
+-- PTU fails with UPDATE_NEEDED status
+--
+-- Note: Script is designed for EXTERNAL_PROPRIETARY flow
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared Libraries ]]
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local testCasesForExternalUCS = require('user_modules/shared_testcases/testCasesForExternalUCS')
+local sdl = require('SDL')
+local commonTestCases = require('user_modules/shared_testcases/commonTestCases')
+local testCasesForPolicySDLErrorsStops = require('user_modules/shared_testcases/testCasesForPolicySDLErrorsStops')
+
+--[[ Local variables ]]
+local grpId = "Location-1"
+local checkedStatus = "UPDATE_NEEDED"
+local checkedSection = "disallowed_by_external_consent_entities_off"
+
+local params = { "entityId", "entityType" }
+local values = {
+  { v = "1", desc = "String" },
+  { v = 1.23, desc = "Float" },
+  { v = {}, desc = "Empty table" },
+  { v = { entityType = 1, entityID = 1 }, desc = "Non-empty table" },
+  { v = -1, desc = "Below Min" },
+  { v = 129, desc = "Beyond Max" },
+  { v = "", desc = "Blank" },
+  { v = nil, desc = "Null" }
+}
+
+--[[ General Precondition before ATF start ]]
+commonFunctions:SDLForceStop()
+commonSteps:DeleteLogsFileAndPolicyTable()
+testCasesForExternalUCS.removePTS()
+
+--[[ General Settings for configuration ]]
+Test = require("user_modules/connecttest_resumption")
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:CheckSDLStatus()
+  testCasesForExternalUCS.checkSDLStatus(self, sdl.RUNNING)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+local n = 1
+local startLine = 1
+
+local function sequence(desc, updateFunc)
+
+  local tcName = "TC[" .. string.format("%02d", n) .. "] - [" .. desc .. "] "
+  tcName = tcName .. string.rep("-", 50 - string.len(tcName))
+
+  Test[tcName] = function() end
+
+  function Test.StopSDL()
```

Comment:
@istoimenova, Agreed, but I tried to perform clean run for every case.
Test consists of 18 cases. Is it possible to perform so many attempts within 1 PTU?
Won't we have a issue with retry strategy etc.?